### ISSUE TITLE
Cyberpunk Red compatbility

### ISSUE THREAD

**Issue by PestoPosta** [Saibot393/LocknKey#105]
Can Cyberpunk Red get some love?

Thanks in advance for the time.
I really hope to be able to use this mod :)

**Comment by Saibot393** [comment]
The basic formula for for example lock picking should look like this for the Cyberpunk RED system: `1d12 + @actor.system.stats.tech.value + @skills.Pick_Lock.system.level`
I am however not very familiar with the system. I would guess that thinks like critical roles and such are lacking, could you give me quick run down of all the relevant system rules and rolls that would be relevant for the modules function?

**Comment by PestoPosta** [comment]
Thanks much, d10s not d12, but I think I can wing it from here. :)
I never know "@actor.system.stats.tech.value" these things format, is it more or less the same from system to system?

1 = crit fail, you roll a d10 again and subtract it from your roll.
10 = crit success, roll a d10 again and add it to your roll. 

Cyberpunk Uncensored does really good tutorials.
https://www.youtube.com/watch?v=D_iQP6-4F3k

And someone was kind enough to make the players hand book actually usable
https://www.worldanvil.com/w/cyberpunk-konigskind/c/player-handbook-category

**Comment by Saibot393** [comment]
Yes the rough format is more or less the same between many systems (though there is technically nothing forcing system developers to abide by that). For D&D5e it would for example look like this `@actor.system.abilities.dex.mod`.

I will look into adding the crit rules with one of the next updates, might take a couple of weeks though.

**Comment by PestoPosta** [comment]
Thanks much.
With technical I am know enough to get 80% there,
but there always seems to be something either finicky or dumb that I just miss. XD
Yeah I am just glad the convention has been established. Its one of the things that
bugs the snot out of me in other modding communities. 

Thanks again
Your a scholar and a gent.

**Comment by PestoPosta** [comment]
Sorry to double post, but is it possible to add multiple formulas?
Because there is also electronics / security tech.

**Comment by Saibot393** [comment]
You can add a third formula type (in addition to Lock Picking and Lock Breaking). To do that, enter the name into the world setting "Custom check name" and the appropiate formula into "Custom check formula". This custom check will work just as the other two and has a seperate lock DC

**Comment by PestoPosta** [comment]
Cool cool that is what I thought that was for.

I assume that the skill is something to the effect of
@actor.system.stats.tech.value + @skills.Electronics/Security_Tech_(x2).system.level

Thanks again :)
This is one of those mods/features I've wanted in a vtt since we move to a vtt as the primary method of play.
So I am really looking forward to fiddling around with it. :)

**Comment by Saibot393** [comment]
Happy to hear that you enjoy the module.

**Comment by Saibot393** [comment]
I just had a closer look at the roll formula for Electronics/Security Tech:
In the old version there was a small problem with the `/` in the name which prevented the formula from working, also the internal name is actually `Electronics/Security_Tech`, i would recommend updating to v3.0.6, where the following formula should work:
`@skills.Electronics_Security_Tech.system.level`

**Comment by PestoPosta** [comment]
Awesome :D

You are rapidly climbing your way up the list of my favorite modders ;)

**Comment by Saibot393** [comment]
I am not really familiar with the system, how/where would these mods be applied? Also, what exactly are you refering to with "base number"?

**Comment by TheMothpriest** [comment]
![image](https://github.com/user-attachments/assets/af6cf37b-4719-4c91-b311-3c77aad6fe04)
The Base number is the total value of an actor's stats (Int , Ref, Dex, Etc) + Skill ( in this example I'm using resist torture and drugs as it has a mod applied to it) +  a ( Plus/minus from a specific condition in this case being pulled from an item.)                            

As far as I can tell all mods are pulled from the effects tab on the sheet. Also in speaking with the system module developers skills are considered items the way it is currently programed from the sounds of things based on what I've seen by looking through the conditions this seems to be the case for most things. I am unsure if that helps diagnose the issue. 

_As an aside_: I also had the initial lockpick formula you posted above working (Which was also apparently curious to the developers of the system module as they apparently didn't use underscores in there system) but as of last night testing it with my players I just hear a roll sound (Likely from dice so nice) but nothing happens ( Not even the chat messages I was receiving during my initial testing before). I hope this is enough to go on and apologize if it is not.

**Comment by Saibot393** [comment]
So to get a value for let's say tech + electronics security + bonus the formula would look something like this:
`@actor.system.stats.tech.value + @skills.Electronics_Security_Tech.system.level + @actor.bonuses.accounting.electronicsAndSecurityTech`
How exactly should do you want the formula to work?

**Comment by TheMothpriest** [comment]
The bonus in most cases would not be from another skill (This assuming the accounting in the formula example for security is not part of the formula expression ) But from an effect mod say +/- 2 from the use of a specific drug or from a critical injury. So Ideally the formula would be expressed as Stat (Tech) + The level in said skill  (Electronic security) + MOD value +/- . In simplest terms I can get the formula to pull the Stat value and Level value but not the MOD value.

**Comment by Saibot393** [comment]
I mistyped the bonus formula, for the specific example it should have been `@actor.bonuses.electronicsAndSecurityTech`. Does that work for you? It seemed to work in my test

**Comment by TheMothpriest** [comment]
Yes that worked perfectly . Thankyou very much for your time. Also great module will probably end up using this quite a bit now

**Comment by Saibot393** [comment]
Glad to hear that

**Comment by TheMothpriest** [comment]
So one last thing (hopefully) I tried to copy the same syntax for the Pick Lock and Pick Pocket skills but that doesn't seem to work how would I write those for the bonuses ? actually seemed Pick_Pocket didn't work at all so probably have that completely wrong. Thank you in advance

**Comment by Saibot393** [comment]
The skills seem to work for me, more specifically, these seem to have the right results:
`1d10 + @actor.system.stats.tech.value + @skills.Pick_Lock.system.level + @actor.bonuses.pickLock`
`1d10 + @actor.system.stats.tech.value + @skills.Pick_Pocket.system.level + @actor.bonuses.pickPocket`

**Comment by TheMothpriest** [comment]
Yup those worked
